app/models/search_parser.rb

```diff
@@ -22,8 +24,12 @@ class SearchParser < Parslet::Parser
 
   # User::VALID_USERNAME
   rule(:commenter) { str("commenter:") >> match("[@~]").repeat(0, 1) >> match("[A-Za-z0-9_\\-]").repeat(1, 24).as(:commenter) >> space? }
-  # reproduce the <domain> named capture in URL_RE
-  rule(:domain) { str("domain:") >> match("[A-Za-z0-9_\\-\\.]").repeat(1).as(:domain) >> space? }
+  rule(:domain) {
+    (
+      (match("[a-z0-9]") >> match("[a-z0-9\\-]").repeat(1, 62) >> str(".")).repeat(1) >>
```

Comment:
Please prefix an optional `str("domain:")` so we keep supporting the old syntax.